### ISSUE TITLE
Cannot connect device Pupil glasses invisible

### ISSUE THREAD

**Issue by khangtt0109** [pupil-labs/realtime-python-api#35]
My code write on python 3.9.16 run the function discover_devices and discover_one_devices cannot dectect the device Pupil glasses invisible.

**Comment by khangtt0109** [comment]
> Hi @khangtt0109 Some networks suppress the auto-discovery mechanism due to security reasons, e.g. university networks.
> 
> Alternatively, you can connect to the device by providing the IP address and port explicitly:
> 
> ```python
> from pupil_labs.realtime_api.simple import Device
>

**Comment by khangtt0109** [comment]
> Hi @khangtt0109 Some networks suppress the auto-discovery mechanism due to security reasons, e.g. university networks.
> 
> Alternatively, you can connect to the device by providing the IP address and port explicitly:
> 
> ```python
> from pupil_labs.realtime_api.simple import Device
> 
> device = Device("192.168.0.2", 8080)
> ```

How do I can check the device ID of pupil glasses ? I tried the ip that 127.0.0.1:50020 but it not works.

**Comment by khangtt0109** [comment]
> 

Can you show me more detail ? I connect the glasses with my PC and I use python code to connect and get real-time data from the glasses but I can't connect the glasses ! By using library realtime-python-api that I tried connect but I have no idea to how to connect.

**Comment by papr** [comment]
@khangtt0109 
> I connect the glasses with my PC

Ah, this might be the source of the misunderstanding. You need to connect the Pupil Invisible glasses to the Companion device (phone) and have the script connect via WIFI to the phone.

**Comment by khangtt0109** [comment]
> 

I can understand that I will connect the Pupil Invisible glasses to the Companion device (phone) and use the script to connect from PC to phone to get realtime data! All that ?

**Comment by papr** [comment]
Yes! If the phone is in the same network as the PC running the script, `discover_devices` should start working, too.

**Comment by khangtt0109** [comment]
One more question, is there any solution that connect Pupil Invisible glasses to PC and PC get and visualize real-time data ?

Thank a lot

**Comment by papr** [comment]
No, the phone is needed to calculate the realtime data.

**Comment by khangtt0109** [comment]
> No, the phone is needed to calculate the realtime data.

Thank you very much! I must set up my device again.

**Comment by papr** [comment]
Happy to help!

**Comment by suvodip1212** [comment]
Hey, Sorry for hijacking the thread. I noticed something weird today. I am getting connected to the pupil companion app through Python. However, instead of getting gaze data, I get the error **"no RTP received for 10 seconds: closing"**. Here is a sample code to reproduce the error:
`
import nest_asyncio
nest_asyncio.apply()
import cv2


from pupil_labs.realtime_api.simple.device import Device

ip = "your companion ip here"

def main():

    # Look for devices. Returns as soon as it has found the first device.

    print("Looking for the next best device...")

    device = Device(address=ip, port="8080") 

    if device is None:

        print("No device found.")

        raise SystemExit(-1)


    print(f"Connecting to {device}...")


    try:

        while True:

            frame, gaze = device.receive_matched_scene_video_frame_and_gaze()

            cv2.circle(

                frame.bgr_pixels,

                (int(gaze.x), int(gaze.y)),

                radius=80,

                color=(0, 0, 255),

                thickness=15,

            )


            cv2.imshow("Scene camera with gaze overlay", frame.bgr_pixels)

            if cv2.waitKey(1) & 0xFF == 27:

                break

    except KeyboardInterrupt:

        pass

    finally:

        print("Stopping...")

        device.close()  # explicitly stop auto-update



if __name__ == "__main__":

    main() `

**Comment by suvodip1212** [comment]
Hey, Sorry for hijacking the thread. I noticed something weird today. I am getting connected to the pupil companion app through Python. However, instead of getting gaze data, I get the error "no RTP received for 10 seconds: closing". Here is a sample code to reproduce the error:
` # -*- coding: utf-8 -*-
import nest_asyncio
nest_asyncio.apply()
import cv2


from pupil_labs.realtime_api.simple.device import Device

ip = "10.0.120.52"

def main():

    # Look for devices. Returns as soon as it has found the first device.

    print("Looking for the next best device...")

    device = Device(address=ip, port="8080") 

    if device is None:

        print("No device found.")

        raise SystemExit(-1)


    print(f"Connecting to {device}...")


    try:

        while True:

            frame, gaze = device.receive_matched_scene_video_frame_and_gaze()

            cv2.circle(

                frame.bgr_pixels,

                (int(gaze.x), int(gaze.y)),

                radius=80,

                color=(0, 0, 255),

                thickness=15,

            )


            cv2.imshow("Scene camera with gaze overlay", frame.bgr_pixels)

            if cv2.waitKey(1) & 0xFF == 27:

                break

    except KeyboardInterrupt:

        pass

    finally:

        print("Stopping...")

        device.close()  # explicitly stop auto-update
if __name__ == "__main__":
    main()
'

**Comment by suvodip1212** [comment]
Hey, Sorry for hijacking the thread. I noticed something weird today. I am getting connected to the pupil companion app through Python. However, instead of getting gaze data, I get the error "no RTP received for 10 seconds: closing". Here is a sample code to reproduce the error: [pupil_invisible.py](https://gist.github.com/suvodip1212/9124e8f1ecea775f8f6a3aa800ec7c0d). I was wondering what could be done to fix this.
